public/components/tooltip/create_tooltip.tsx

```diff
@@ -1,25 +1,26 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import { Popup, MapGeoJSONFeature } from 'maplibre-gl';
+import { Popup, MapGeoJSONFeature, LngLat } from 'maplibre-gl';
 
 import { MapLayerSpecification, DocumentLayerSpecification } from '../../model/mapLayerType';
 import { FeatureGroupItem, TooltipContainer } from './tooltipContainer';
 
-type Options = {
+interface Options {
   features: MapGeoJSONFeature[];
   layers: DocumentLayerSpecification[];
   showCloseButton?: boolean;
   showPagination?: boolean;
   showLayerSelection?: boolean;
-};
+}
 
 export function isTooltipEnabledLayer(
   layer: MapLayerSpecification
 ): layer is DocumentLayerSpecification {
   return (
     layer.type !== 'opensearch_vector_tile_map' &&
     layer.type !== 'custom_map' &&
-    layer.source.showTooltips === true
+    layer.source.showTooltips === true &&
+    !!layer.source.tooltipFields?.length
```

Comment:
Will the last condition `!!layer.source.tooltipFields?.length` returns false if tooltipFields are not provided after enabling toolTips?

Reply:
showTooltips is a flag where user controls whether to show toolTip or not, however, tooltipFields contains list of tooltip fields. The last condition checks whether length of tooltipFields is > 0 or not

Reply:
Yes, I got it. But, I'm trying to understand what will be the value of `!!layer.source.tooltipFields?.length` if no tooltipFields are provided. If it is not an optional field it's value will be false. But, what will be the value when it is not optional?

Reply:
false